src/api/c/pinverse.cpp

```diff
@@ -0,0 +1,170 @@
+/*******************************************************
+ * Copyright (c) 2014, ArrayFire
+ * All rights reserved.
+ *
+ * This file is distributed under 3-clause BSD license.
+ * The complete license agreement can be obtained at:
+ * http://arrayfire.com/licenses/BSD-3-Clause
+ ********************************************************/
+
+#include <backend.hpp>
+#include <Array.hpp>
+
+#include <af/array.h>
+#include <af/complex.h>
+#include <af/defines.h>
+#include <af/lapack.h>
+#include <arith.hpp>
+#include <blas.hpp>
+#include <cast.hpp>
+#include <common/ArrayInfo.hpp>
+#include <common/err_common.hpp>
+#include <diagonal.hpp>
+#include <handle.hpp>
+#include <join.hpp>
+#include <logic.hpp>
+#include <math.hpp>
+#include <reduce.hpp>
+#include <select.hpp>
+#include <svd.hpp>
+#include <transpose.hpp>
+
+using af::dim4;
+using af::dtype_traits;
+using std::vector;
+using std::swap;
+
+using namespace detail;
+
+const double dfltTol = 1e-6;
+
+template<typename T>
+Array<T> pinverseSvd(const Array<T> &in, const double tol)
+{
+    // Moore-Penrose Pseudoinverse
+
```

Comment:
Nit: Extra line

Reply:
Here I intentionally put an extra line because the comment doesn't apply only to the code block right below it, but rather the whole function. Maybe I should put this comment above the signature instead eh?

Reply:
It would be better to apply that comment to the whole function. Maybe move it above the function